### PR TITLE
Feature/add friendships

### PR DESCRIPTION
**Pull request by iam-Akshat**
- Make friendships model
- Add functionality to send and accept requests

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @iam-Akshat  :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] There are 2 users, current_user and other_user. As the current_user, on clicking on the `See profile` button in front of other_user, I got the error below.
image
* [x] The `SEND REQUEST` button isn't working. Kindly fix it.
* [x] User can see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
* [x] You said you weren't sure if you're to add tests.
  * [ ] You're expected to add unit and integration tests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

### Make the requested changes and submit for another review.
## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

File: app/models/user.rb
Comment (ghost) [inline app/models/user.rb]: Add tests for the usage of options `dependent`, `foreign_key` and `through`.

Example line 10, your test should be, ` it { should have_many(:comments).class_name('Comment').dependent(:destroy)}`.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑

### Hello team 👋,

**Great job implementing the requested changes 👏. However, a few fixes are still needed.**

1). **Linter checks are not showing on your pull request.**

2). **Users should not see each other's posts unless they are confirmed friends.**

3). **Say, Mercy sends Jen a friend request. When Jen logs in, she should not see a friend request button next to Mercy's name and her profile.**

4). **Add tests for your views and helpers.**

5). **Please see on more in the comment below.**

## Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone. Everything seems to be working :clap: 
However, some changes are required before this milestone can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] As mentioned by the previous TSE, your GitHub Actions linter checks are not running on your PR page. 
- Deleting the entire content of .github/workflows/linters.yml and recreating the content might fix this problem. You can find the configuration [here](https://github.com/microverseinc/linters-config/tree/master/ror).
- If you've tried all you can but you're still experiencing the same issue, you can reach out to [a full-time TSE](https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-).
- Ensure that all linter checks run successfully before you request another review.

**[ OPTIONAL CHANGES ]**

- [ ] You do not have to show other users' posts on their profile show page. This might be a nice feature for the currently logged-in user, but not for other users.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by Georjane — APPROVED** [review]
## Status: PR Approved 💯 🥇 
### Hi there team ✋🏽 
Nice, you successfully set up the GitHub actions for linters. Your project meets the minimum requirements for approval. You can go ahead and merge this PR.

Good luck on the next milestone and happy coding 💻

**Pull request merged**
